instrumentation/jdbi3/src/test/java/brave/jdbi3/ITJdbi3BravePlugin.java

```diff
@@ -0,0 +1,150 @@
+    protected static final Tracing tracing = Tracing.newBuilder()
+        .currentTraceContext(StrictCurrentTraceContext.create())
+        .addSpanHandler(spans).build();
+    static final String QUERY = "select 'hello world'";
+    static final String ERROR_QUERY = "select unknown_field FROM unknown_table";
+    protected static Jdbi jdbi = null;
+
+    static int envOr(String key, int fallback) {
+        return System.getenv(key) != null ? Integer.parseInt(System.getenv(key)) : fallback;
+    }
+
+    static String envOr(String key, String fallback) {
+        return System.getenv(key) != null ? System.getenv(key) : fallback;
+    }
+
+    public Jdbi buildJdbi() {
+        StringBuilder url = new StringBuilder("jdbc:mysql://");
+        url.append(envOr("MYSQL_HOST", "127.0.0.1"));
+        url.append(":").append(envOr("MYSQL_TCP_PORT", 3306));
+        String db = envOr("MYSQL_DB", null);
+        if (db != null) {
+            url.append("/").append(db);
+        }
+        url.append("?zipkinServiceName=").append("myservice");
+        url.append("&serverTimezone=").append("UTC");
+
+        MysqlDataSource dataSource = new MysqlDataSource();
+        dataSource.setUrl(url.toString());
+
+        dataSource.setUser(System.getenv("MYSQL_USER"));
+        System.err.println(dataSource.getUser());
+        assumeTrue(dataSource.getUser() != null,
+            "Minimally, the environment variable MYSQL_USER must be set");
+        dataSource.setPassword(envOr("MYSQL_PASS", ""));
+
+        Jdbi mysql = Jdbi.create(dataSource);
+        mysql.installPlugin(Jdbi3BravePlugin.newBuilder(tracing).includeBindVariables(true).build());
+        return mysql;
+    }
+
+    @Test
+    void reportsServerAddressForMysqlDatabases() {
+        prepareExecuteSelect(jdbi, QUERY);
+
+        assertThat(spans)
+            .extracting(MutableSpan::remoteServiceName)
+            .contains("myservice");
+    }
+
+    @BeforeEach
+    void init() {
+        if (jdbi != null) {
+            return;
+        }
+        jdbi = buildJdbi();
+    }
+
+    @BeforeEach
+    void clearSpans() {
+        spans.clear();
+    }
+
+    @AfterEach
+    void close() {
+        tracing.close();
+    }
+
+    @Test
+    void makesChildOfCurrentSpan() {
+        ScopedSpan parent = tracing.tracer().startScopedSpan("test");
+        try {
+            prepareExecuteSelect(jdbi, QUERY);
+        } finally {
+            parent.finish();
+        }
+
+        assertThat(spans)
+            .hasSize(2);
+    }
+
+    @Test
+    void reportsClientKindToZipkin() {
+        prepareExecuteSelect(jdbi, QUERY);
+
+        assertThat(spans)
+            .extracting(MutableSpan::kind)
+            .containsExactly(CLIENT);
+    }
+
+    @Test
+    void defaultSpanNameIsOperationName() {
+        prepareExecuteSelect(jdbi, QUERY);
+
+        assertThat(spans)
+            .extracting(MutableSpan::name)
+            .containsExactly("Query");
```

Comment:
I think you've misunderstood me. We shouldn't model differently than mysql. Please match your tests to what is here, as the span name shouldn't be "Query" that's not a SQL operator

https://github.com/openzipkin/brave/blob/ff4006230706a8ee357bc6ba18479faa2c3a86c2/instrumentation/mysql8/src/test/java/brave/mysql8/ITTracingQueryInterceptor.java#L109

Reply:
That´s the reason I had the `jdbi.` prefix originally - the name is the Jdbi description of the statement type, not the first word of the statement. Extracting the first word of the statement seems very brittle (e.g. it fails when using comment prefixes, which is generally is good practice), and since the full SQL is in the `sql.query` tag, it is also redundant. Using the Jdbi description of the type of statement (e.g. Query, Update, or Call) seems to me to be a better way.
I can change it if you insist, but I do think it would result in a worse implementation.